### PR TITLE
Development

### PR DESCRIPTION
**Pull request by viniciusou**
Development of Upcoming Movies App MVP

### CODE REVIEW

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/Models/Movie.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UpcomingMovies.Models
+{
+     public class Movie: ObservableObject
+     {
+        public string Title { get; set; }
+
+        [JsonProperty(PropertyName = "genre_ids")]
```
Comment (arctouch-victorbotamedi) [inline UpcomingMovies/UpcomingMovies/UpcomingMovies/Models/Movie.cs:15]: You can use a `ContractResolver`, so you don't need to put serialization code in your models

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/Settings/Settings.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace UpcomingMovies.Settings
+{
+    public static class Settings
+    {
+        public static string ApiKey { get; } = "1f54bd990f1cdfb230adb312546d765d";
```
Comment (arctouch-victorbotamedi) [inline UpcomingMovies/UpcomingMovies/UpcomingMovies/Settings/Settings.cs:11]: Just a syntax sugar :), you can write  `ApiKey => "1f54bd990f1cdfb230adb312546d765d"

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/ViewModels/MovieListViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+                if (response.IsSuccessStatusCode)
+                {
+                    var genrest = new List<Genre>();
+                    var json = await response.Content.ReadAsStringAsync();
+
+                    var jObject = JObject.Parse(json);
+
+                    var movies = jObject.SelectToken("results").ToObject<List<Movie>>(); ;
+                    TotalResults = (int)jObject.SelectToken("total_results");
```
Comment (arctouch-victorbotamedi) [inline UpcomingMovies/UpcomingMovies/UpcomingMovies/ViewModels/MovieListViewModel.cs:126]: You could have an intermediate API model to make this call simpler using `JsonConvert.DeserializeObject<T>`

**Review by arctouch-victorbotamedi — COMMENTED** [review]
Good work, simple and easy to understand.
Some points of attention are:
* Error handling
* Could have a layer to communicate with the API to separate this logic from the view models
* Could have used the search API :)
* Smaller more meaningful methods

**Comment by viniciusou** [comment]
@arctouch-victorbotamedi 

Thank you for your feedback. These suggestions will make the quality of my code evolve a lot.

**"Error handling"** 
It's not gonna be forgotten again :)

**"Could have a layer to communicate with the API to separate this logic from the view models"**
Great, it improves the separation of concerns.

**"Could have used the search API"**
The API was not very accurate to retrieve only upcoming movies. I've also tried the discover API, but the best results were achieved using the author's suggestion for the upcoming API to avoid outdated movies. I will give a try on the search API.  

**Smaller more meaningful methods**
The suggested API communication layer will help on that.

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/Models/Movie.cs (hunk 1 above)
Comment (viniciusou) [reply UpcomingMovies/UpcomingMovies/UpcomingMovies/Models/Movie.cs:15]: Nice, it makes classes cleaner.

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/Settings/Settings.cs (hunk 2 above)
Comment (viniciusou) [reply UpcomingMovies/UpcomingMovies/UpcomingMovies/Settings/Settings.cs:11]: Awesome :)

File: UpcomingMovies/UpcomingMovies/UpcomingMovies/ViewModels/MovieListViewModel.cs (hunk 3 above)
Comment (viniciusou) [reply UpcomingMovies/UpcomingMovies/UpcomingMovies/ViewModels/MovieListViewModel.cs:126]: I got stuck on how to use JsonConvert.DeserializeObject<T> in this case, that's why I've used JObject. I will check how to implement that.

**Pull request merged**